File: 1496-lucky-numbers-in-a-matrix/1496-lucky-numbers-in-a-matrix.go
func luckyNumbers (matrix [][]int) []int {
    n := len(matrix)
    m := len(matrix[0])
    mininRow := make([]int, n)
    maxinCol := make([]int, m)
    var lucky []int 
    for i := 0; i < n; i++{
        minn := 1000000
        for j := 0; j < m; j++{
            if minn > matrix[i][j]{
                minn = matrix[i][j]
            }
        }
        mininRow[i] = minn
    }

    for i := 0; i < m; i++{
        maxx := 0
        for j := 0; j < n; j++{
            if maxx < matrix[j][i]{
                maxx = matrix[j][i]
            }
        }
        maxinCol[i] = maxx
    }

    for _, num := range mininRow{
        for _, maxNum := range maxinCol {
            if num == maxNum{
                lucky = append(lucky, num)
            }
        }
    }

    return lucky
}

File: 1496-lucky-numbers-in-a-matrix/1496-lucky-numbers-in-a-matrix.py
class Solution:
    def luckyNumbers (self, matrix: List[List[int]]) -> List[int]:
        mininRow = []
        maxinCol = []
        lucky = []
        for i in range(len(matrix)):
            mininRow.append(min(matrix[i]))
            maxx = 0
        
        for i in range(len(matrix[0])):
            maxx = 0
            for j in range(len(matrix)):
                maxx = max(maxx, matrix[j][i])
            maxinCol.append(maxx)
            maxx = 0
        # print(maxinCol)
        # print(mininRow, maxinCol)
        for num in mininRow:
            if num in maxinCol:
                lucky.append(num)
            
        
        return lucky